builds/fabric8-che/assembly/user-auth-valve/src/main/java/com/redhat/che/valve/UserAuthValve.java

```diff
@@ -113,36 +122,16 @@ private String getToken(Request request) {
     }
 
     public boolean isKeycloakDisabled() {
-        retrieveKeycloakDisabledSetting();
-        return keycloakDisabled;
-    }
-    
-    public void retrieveKeycloakDisabledSetting() {
-        if (! keycloakDisabledRetrieved) {
-            URL url;
-            HttpURLConnection conn;
-            try {
-                url = new URL(KEYCLOAK_SETTINGS_ENDPOINT);
-                conn = (HttpURLConnection) url.openConnection();
-                conn.setRequestMethod("GET");
-                BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
-                StringBuilder response = new StringBuilder();
-                String line;
-                while ((line = in.readLine()) != null) {
-                    response.append(line);
-                }
-                if (LOG.isInfoEnabled()) {
-                    LOG.info("KeycloakSettings = " + response);
-                }
-                Matcher matcher = DISABLED_SETTING_PATTERN.matcher(response.toString());
-                if (matcher.matches()) {
-                    String value = matcher.group(1);
-                    keycloakDisabled = "true".equals(value);
-                    keycloakDisabledRetrieved = true;
-                }
-            } catch (IOException e) {
-                LOG.error("Exception during Keycloak settings retrieval", e);
-            }
+        if (keycloakDisabled != null) {
+            return keycloakDisabled;
+        }
+        
+        Map<String, String> keycloakSettings = KeycloakSettings.get();
+        if (keycloakSettings == null) {
+            LOG.warn("KeycloakSettings = null => don't disable Keycloak");
+        } else {
+            keycloakDisabled = "true".equals(keycloakSettings.get(DISABLED_SETTING));
```

Comment:
could we use https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#parseBoolean(java.lang.String)